hazelcast/src/main/java/com/hazelcast/nio/Packet.java

```diff
@@ -38,33 +39,61 @@
 
     public static final byte VERSION = 4;
 
+
+    //             PACKET HEADER FLAGS
+    //
+    // Flags are dispatched against in a cascade:
+    // 1. URGENT (bit 4)
+    // 2. Packet type (bits 0, 2, 5)
+    // 3. Flags specific to a given packet type (bits 1, 6)
+
+
+    // 1. URGENT flag
+
+    /** Marks the packet as Urgent  */
+    public static final int FLAG_URGENT = 1 << 4;
+
+
+    // 2. Packet type flags, encode up to 8 packet types. Historically the first three packet
+    // types were encoded as separate, mutually exclusive flags. These are given below.
+    // The enum Packet.Type should be used to encode/decode the type from the header flags bitfield.
+
+    /** Packet type bit 0. Historically the OPERATION type flag. */
     public static final int FLAG_OP = 1 << 0;
-    public static final int FLAG_RESPONSE = 1 << 1;
+    /** Packet type bit 1. Historically the EVENT type flag. */
     public static final int FLAG_EVENT = 1 << 2;
-    public static final int FLAG_URGENT = 1 << 4;
+    /** Packet type bit 2. Historically the BIND type flag. */
     public static final int FLAG_BIND = 1 << 5;
 
-    /**
-     * A flag to indicate this is a special control packet for the operation system like invocation-heartbeats.
-     */
+    // 3. Type-specific flags. Same bits can be reused within each type
+
+    // 3.a Operation packet flags
+
+    /** Marks an Operation packet as Response */
+    public static final int FLAG_OP_RESPONSE = 1 << 1;
+    /** Marks an Operation packet as Operation control (like invocation-heartbeats) */
     public static final int FLAG_OP_CONTROL = 1 << 6;
 
-    /**
-     * Flag to indicate this will be a Jet packet
-     */
-    public static final int FLAG_JET = 1 << 7;
+
+    // 3.b Jet packet flags
+
+    /** Marks a Jet packet as Flow control */
+    public static final int FLAG_JET_FLOW_CONTROL = 1 << 1;
```

Comment:
so this is a JET-packet-specific flag, but there is no JET packet definition. I guess it would deserve an explanation too.